### PR TITLE
Add Template DataPlane project

### PR DESCRIPTION
**Pull request by weshaggard**
- Also adds some wrapper props/targets to enable dataplane builds.
- Update client.yml to use dotnet build commands for both windows and linux
- Add netfx compile support on non-windows

### CODE REVIEW

**Comment by weshaggard** [comment]
cc @Azure/azure-sdk-eng

**Comment by weshaggard** [comment]
@shahabhijeet this change is ready for merging when you get a chance to have a look.

File: dirs.proj
Comment (kurtzeborn) [inline dirs.proj]: Remember to rename these to "data-plane"

File: src/SDKs/Template/data-plane/Microsoft.Azure.Template/Microsoft.Azure.Template.csproj
Comment (kurtzeborn) [inline src/SDKs/Template/data-plane/Microsoft.Azure.Template/Microsoft.Azure.Template.csproj]: AssemblyName is redundant.  It's defaulted from the Project name.

File: .azure-pipelines/client.yml
Comment (kurtzeborn) [inline .azure-pipelines/client.yml]: We should parameterize scope here probably.

**Review by kurtzeborn — APPROVED** [review]

**Comment by weshaggard** [comment]
I'm not until next Tues so I'd like to hold off on merging this until then so that we can closely follow it with the changes for keyvault. In the mean time I'd still appreciate a review from @shahabhijeet.

File: eng/Directory.Build.Data.targets
Comment (weshaggard) [inline eng/Directory.Build.Data.targets]: Need to setup a strong name key for the data-plane libraries.

File: Directory.Build.props
Comment (shahabhijeet) [inline Directory.Build.props]: @weshaggard maybe this is a style issue
Can we not add trailing directory separator for root level properties/paths
for e.g. rather than adding trailing directory separator for <RepoEngPath>, use the directory separator below
$(RepoEngPath)/Directory.Build.Mgmt.props

Here are my reasons for it
When I use a directory separator, I know for sure that $(RepoEngPath) is a directory and not some other value.

File: dirs.proj
Comment (shahabhijeet) [inline dirs.proj]: @weshaggard what does this do? I thought we were doing multiple directory.build.props files for providing common references

**Review by shahabhijeet — CHANGES_REQUESTED** [review]
Looks good.
I will stop by and understand the reason for
Template
data-plane-*.proj (I thought we were planning to do multiple directory.build.props files

**Review by shahabhijeet — APPROVED** [review]

File: .azure-pipelines/client.yml
Comment (weshaggard) [reply .azure-pipelines/client.yml]: This will likely need to be a follow-up change so we can correctly setup the build pipeline.

**Pull request merged**
